### DISCUSSION TITLE
Problems installing pybamm from conda forge

### DISCUSSION THREAD

**Discussion by AlvarmenTUD** [pybamm-team/PyBaMM#2668] [General]
Hi all, 

I am trying to install PyBaMM from Anaconda Prompt using 
conda install -c conda-forge pybamm

I have created a new environment for this purpose with both python 3.9 and 3.8 but still not able, I am getting the following error: 

![image](https://user-images.githubusercontent.com/122776022/217249882-57565e02-a7b3-4133-8f5a-ec73eab4295b.png)

Been trying to install the package for some time but I am getting this or other problems, doing it from anaconda navigator and from anaconda prompt. 

I have tried with pip from windows powershell and still not able to install the package. 

![image](https://user-images.githubusercontent.com/122776022/218031781-7c4143e5-c27e-4565-b63a-5550e483f708.png)

Any idea what can be the cause of these problems???
Thanks a lot in advance.

**Comment by Jerry841124** [comment]
Use python 3.7 today, pip install is OK

**Comment by Saransh-cpp** [comment]
@AlvarmenTUD please install `wheel` before installing `pybamm` using `pip` -
```
pip install wheel
pip install pybamm
```

@Jerry841124 please avoid using `pybamm` on Python `3.7`. `pybamm`'s latest releases do not support Python `3.7`; hence, `pip` will fetch you an outdated version of `pybamm` on Python `3.7`.

**Reply by Jerry841124** [reply]
@Saransh-cpp  May I ask which stable version is recommended?   Thanks!

**Reply by Saransh-cpp** [reply]
The latest pybamm version should usually be preferred - pybamm follows [CalVer](https://github.com/pybamm-team/PyBaMM#versioning). You can use Python 3.8 or 3.9 with pybamm.

**Comment by AlvarmenTUD** [comment]
Dear all, 

I had the latest python version (3.10) and was trying to install the newest version of pybamm (23). It appears that was the problem, not sure if it was something about the library itself or my computer. 

I solved it by creating a new environment with python 3.9 and installing pybamm 22.12. With older versions in both, I was able to install and run it.

**Comment by brosaplanella** [comment]
Hi! PyBaMM still does not support Python 3.10, only 3.8 and 3.9. The most recent versions should work with Python 3.9 as well.

**Comment by pavithraes** [comment]
Hi! I believe PyBaMM now does support Python 3.10, but I'm still having trouble installing the latest version `23.5` using conda-forge.

I've tested on MacOS (x64) and Linux, and conda-forge tends to pick version `0.2.3.post1`.

Minimal example: `conda create -n pybamm-exp -c conda-forge python=3.10 pybamm=23.5`

Traceback:

```python-traceback
UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package python conflicts for:
python=3.10
pybamm=23.5 -> python[version='>=3.8,<3.9.0a0|>=3.9,<3.10.0a0']
pybamm=23.5 -> anytree[version='>=2.4.3'] -> python[version='*|2.7.*|3.5.*|3.6.*|>=3.6|>=3.10,<3.11.0a0|>=3.11,<3.12.0a0|>=3|>=3.7,<3.8.0a0|>=3.12,<3.13.0a0|>=3.6,<3.7.0a0|>=2.7,<2.8.0a0|>=3.12.0rc3,<3.13.0a0|>=3.7.1,<3.8.0a0|3.8.*|>=3.8|>=3.7|>=2.7|3.4.*|>=3.5,<3.6.0a0|>=3.9|>=3.5.3|>=3.5|3.9.*',build=*_cpython]The following specifications were found to be incompatible with your system:

  - feature:/osx-64::__osx==10.16=0
  - feature:/osx-64::__unix==0=0
  - feature:|@/osx-64::__osx==10.16=0
  - feature:|@/osx-64::__unix==0=0
  - pybamm=23.5 -> matplotlib-base[version='>=2.0'] -> __osx[version='>=10.11|>=10.12|>=10.9']
  - pybamm=23.5 -> sympy[version='>=1.8'] -> __unix

Your installed version is: 10.16
```

However, the following options work:
- python 3.9 & pybamm 23.5 with conda-forge
- python 3.10 & pybamm 23.5 with pip

**Reply by Saransh-cpp** [reply]
Thanks for reporting this! I believe https://github.com/conda-forge/pybamm-feedstock/blob/main/recipe/meta.yaml will require a few updates with the upcoming release too. Opening an issue for this.
